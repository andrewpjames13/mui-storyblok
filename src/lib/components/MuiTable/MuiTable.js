import React from 'react';
import PropTypes from 'prop-types';
import { Table } from '@material-ui/core';
import StoryBlok from '../../utils/Storyblok';

import MuiTableHead from './components/MuiTableHead/MuiTableHead';
import MuiTableBody from './components/MuiTableBody/MuiTableBody';

const MuiTable = ({
  rootClass,
  padding,
  size,
  stickyHeader,
  tHead,
  tBody,
}) => {
  const styles = StoryBlok.arrayToMuiStyles(rootClass);
  console.log('table head: ', tHead);

  return (
    <Table
      className={styles.root}
      padding={padding}
      size={size}
      stickyHeader={stickyHeader}
    >
      <MuiTableHead {...tHead[0]} />
      <MuiTableBody {...tBody[0]} />
    </Table>
  );
};

export default MuiTable;

MuiTable.propTypes = {
  /**
   * stroyblok multiselect of css classes
   * Mui Override or extend the styles applied to the component.
   */
  rootClass: PropTypes.arrayOf(PropTypes.string),
  /**
   * mui props: 'default' | 'checkbox' | 'none'
   * Allows TableCells to inherit padding of the Table.
   */
  padding: PropTypes.string,
  /**
   * mui props: 'small' | 'medium'
   * Allows TableCells to inherit size of the Table.
   */
  size: PropTypes.string,
  /**
   * mui props: true | false
   * set the header sticky. (It doesn't work with IE 11)
   */
  stickyHeader: PropTypes.bool,

  /**  MuiTableHead limit of 1 */
  tHead: PropTypes.arrayOf(
    PropTypes.shape({
      component: PropTypes.string.isRequired,
    }),
  ).isRequired,
  /**  MuiTableBody limit of 1 */
  tBody: PropTypes.arrayOf(
    PropTypes.shape({
      component: PropTypes.string.isRequired,
    }),
  ).isRequired,
};

MuiTable.defaultProps = {
  padding: 'default',
  size: 'medium',
  stickyHeader: false,
  rootClass: [],
};
