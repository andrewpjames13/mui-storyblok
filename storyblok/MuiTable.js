module.exports = {
  "name": "MuiTable",
  "display_name": null,
  "updated_at": "2020-04-21T17:26:30.043Z",
  "schema": {
    "rootClass": {
      "type": "options",
      "default_value": "* stroyblok multiselect of css classes \n* Mui Override or extend the styles applied to the component."
    },
    "padding": {
      "type": "option",
      "use_uuid": true,
      "options": [
        {
          "value": "default",
          "name": "default"
        },
        {
          "value": "checkbox",
          "name": "checkbox"
        },
        {
          "value": "none",
          "name": "none"
        }
      ],
      "required": true,
      "default_value": "default",
      "description": "   * mui props: 'default' | 'checkbox' | 'none'\n   * Allows TableCells to inherit padding of the Table."
    },
    "size": {
      "type": "option",
      "use_uuid": true,
      "required": true,
      "options": [
        {
          "value": "small",
          "name": "small"
        },
        {
          "value": "medium",
          "name": "medium"
        }
      ],
      "description": "   * mui props: 'small' | 'medium'\n   * Allows TableCells to inherit size of the Table.",
      "default_value": "medium"
    },
    "stickyHeader": {
      "type": "boolean",
      "use_uuid": true,
      "description": "   * mui props: true | false\n   * set the header sticky. (It doesn't work with IE 11)"
    },
    "tHead": {
      "type": "bloks",
      "restrict_components": true,
      "component_whitelist": [
        "MuiTableHead"
      ],
      "required": true,
      "maximum": "1"
    },
    "tBody": {
      "type": "bloks",
      "required": true,
      "maximum": "1",
      "restrict_components": true,
      "component_whitelist": [
        "MuiTableBody"
      ]
    }
  },
  "image": null,
  "preview_field": null,
  "is_root": false,
  "preview_tmpl": null,
  "is_nestable": true,
  "all_presets": [],
  "preset_id": null,
  "real_name": "MuiTable",
  "component_group_uuid": null
}