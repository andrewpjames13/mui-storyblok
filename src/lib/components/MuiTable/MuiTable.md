MuiTable example:

```js noeditor
import MuiTable from './MuiTable';
```

```js
  const tHead = [{
    component: 'MuiTableHead',
    content: [{
      component: 'MuiTableRow',
      content: [{
        component: 'MuiTableCell',
        size: 'small',
        content: [{
          component: 'MuiTypography',
          content: [{
            component: 'MuiText',
            text: 'Table Header 1',
          }],
        }],
      },
      {
        component: 'MuiTableCell',
        size: 'small',
        content: [{
          component: 'MuiTypography',
          content: [{
            component: 'MuiText',
            text: 'Table Header 2',
          }],
        }],
      },
      {
        component: 'MuiTableCell',
        size: 'small',
        content: [{
          component: 'MuiTypography',
          content: [{
            component: 'MuiText',
            text: 'Table Header 3',
          }],
        }],
      }],
    }],
  }];


  const tBody = [
    {
      component: 'MuiTableBody',
      content: [
        {
          component: 'MuiTableRow',
          content: [
            {
              component: 'MuiTableCell',
              size: 'small',
              content: [
                {
                  component: 'MuiTypography',
                  content: [
                    {
                      component: 'MuiText',
                      text: 'Table Body 1',
                    },
                  ],
                },
              ],
            },
            {
              component: 'MuiTableCell',
              size: 'small',
              content: [
                {
                  component: 'MuiTypography',
                  content: [
                    {
                      component: 'MuiText',
                      text: 'Table Body 2',
                    },
                  ],
                },
              ],
            },
            {
              component: 'MuiTableCell',
              size: 'small',
              content: [
                {
                  component: 'MuiTypography',
                  content: [
                    {
                      component: 'MuiText',
                      text: 'Table Body 3',
                    },
                  ],
                },
              ],
            },
          ],
        },
        {
          component: 'MuiTableRow',
          content: [
            {
              component: 'MuiTableCell',
              size: 'small',
              content: [
                {
                  component: 'MuiTypography',
                  content: [
                    {
                      component: 'MuiText',
                      text: 'Table Body 1',
                    },
                  ],
                },
              ],
            },
            {
              component: 'MuiTableCell',
              size: 'small',
              content: [
                {
                  component: 'MuiTypography',
                  content: [
                    {
                      component: 'MuiText',
                      text: 'Table Body 2',
                    },
                  ],
                },
              ],
            },
            {
              component: 'MuiTableCell',
              size: 'small',
              content: [
                {
                  component: 'MuiTypography',
                  content: [
                    {
                      component: 'MuiText',
                      text: 'Table Body 3',
                    },
                  ],
                },
              ],
            },
          ],
        },
        {
          component: 'MuiTableRow',
          content: [
            {
              component: 'MuiTableCell',
              size: 'small',
              content: [
                {
                  component: 'MuiTypography',
                  content: [
                    {
                      component: 'MuiText',
                      text: 'Table Body 1',
                    },
                  ],
                },
              ],
            },
            {
              component: 'MuiTableCell',
              size: 'small',
              content: [
                {
                  component: 'MuiTypography',
                  content: [
                    {
                      component: 'MuiText',
                      text: 'Table Body 2',
                    },
                  ],
                },
              ],
            },
            {
              component: 'MuiTableCell',
              size: 'small',
              content: [
                {
                  component: 'MuiTypography',
                  content: [
                    {
                      component: 'MuiText',
                      text: 'Table Body 3',
                    },
                  ],
                },
              ],
            },
          ],
        },
      ],
    },
  ];

  <MuiTable
    tHead={tHead}
    tBody={tBody}
    size='small'
  />
```