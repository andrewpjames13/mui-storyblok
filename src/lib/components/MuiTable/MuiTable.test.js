import React from 'react';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import MuiTable from './MuiTable';

function setup() {
  const props = {
    padding: '4px',
    size: 'medium',
    sitckyHeader: false,
    tHead: [
      {
        content: [
          {
            component: 'MuiTableRow',
            content: [
              {
                component: 'MuiTableCell',
                content: [
                  {
                    component: 'MuiTypography',
                    content: [
                      {
                        component: 'MuiText',
                        text: 'wooo',
                      },
                    ],
                  },
                ],
              },
            ],
          },
        ],
      },
    ],
    tBody: [{
      content: [{
        hover: false,
        content: [{
          size: 'medium',
          align: 'center',
          content: [{
            align: 'inherit',
            color: 'inherit',
            width: '100%',
            height: '100%',
            content: [{
              text: 'table body 1',
              component: 'MuiText',
            }],
            variant: 'body1',
            component: 'MuiTypography',
          }],
          padding: 'default',
          component: 'MuiTableCell',
          sortDirection: 'asc',
        }],
        selected: false,
        component: 'MuiTableRow',
      }],
      component: 'MuiTableBody',
    }],
  };
  const comp = shallow(<MuiTable {...props} />);
  return { comp, props };
}

describe('<MuiTable />', () => {
  test('snapshot', () => {
    const { props } = setup();
    const tree = renderer.create(
      <MemoryRouter>
        <MuiTable {...props} />
      </MemoryRouter>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should render MuiTable', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });
});
