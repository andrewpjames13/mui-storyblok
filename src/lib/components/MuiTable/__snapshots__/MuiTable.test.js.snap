// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MuiTable /> snapshot 1`] = `
<table
  className="MuiTable-root makeStyles-root-2"
  role={null}
>
  <thead
    className="MuiTableHead-root makeStyles-root-5"
    role={null}
  >
    <tr
      className="MuiTableRow-root makeStyles-root-7 MuiTableRow-head"
      role={null}
    >
      <th
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-head makeStyles-root-13"
        scope="col"
      >
        <div
          className="MuiTypography-root makeStyles-root-26 MuiTypography-body1"
        >
          <span
            className="MuiTypography-root makeStyles-root-27 MuiTypography-body1"
          >
             wooo 
          </span>
        </div>
      </th>
    </tr>
  </thead>
  <tbody
    className="MuiTableBody-root makeStyles-root-58"
    role={null}
  >
    <tr
      className="MuiTableRow-root makeStyles-root-60"
      role={null}
    >
      <td
        aria-sort="ascending"
        className="MuiTableCell-root MuiTableCell-body makeStyles-root-61 MuiTableCell-alignCenter"
      >
        <div
          className="MuiTypography-root makeStyles-root-62 MuiTypography-body1 MuiTypography-colorInherit"
        >
          <span
            className="MuiTypography-root makeStyles-root-63 MuiTypography-body1"
          >
             table body 1 
          </span>
        </div>
      </td>
    </tr>
  </tbody>
</table>
`;
